lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -858,7 +858,21 @@ void PatchEntryPointMutate::setFuncAttrs(Function *entryPoint) {
     builder.addAttribute("amdgpu-depth-export", hasDepthExport ? "1" : "0");
 
     // mmSPI_SHADER_COL_FORMAT is used for fully compiled shaders
-    unsigned colFormat = m_pipelineState->getPalMetadata()->getRegister(mmSPI_SHADER_COL_FORMAT);
+    unsigned colFormat = 0;
+    if (m_pipelineState->useRegisterFieldFormat()) {
+      auto &colFormatNode = m_pipelineState->getPalMetadata()
+                                ->getPipelineNode()
+                                .getMap(true)[Util::Abi::PipelineMetadataKey::GraphicsRegisters]
+                                .getMap(true)[Util::Abi::GraphicsRegisterMetadataKey::SpiShaderColFormat]
+                                .getMap(true);
+      for (auto iter = colFormatNode.begin(); iter != colFormatNode.end(); ++iter) {
+        colFormat = iter->second.getUInt();
+        if (colFormat != EXP_FORMAT_ZERO)
+          break;
+      }
+    } else {
+      colFormat = m_pipelineState->getPalMetadata()->getRegister(mmSPI_SHADER_COL_FORMAT);
+    }
```

Comment:
The meaning of colFormat is different between those two cases. Can you please avoid that by instead doing something like:
```c++
bool hasColorExport = false;
... set to true based on  SpiShaderColFormat / mmSPI_SHADER_COL_FORMAT settings ...
... set to true based on getColorExportCount ...
```

Reply:
Thanks. Fixed.